stl/inc/iterator

```diff
@@ -800,9 +799,25 @@ public:
         _Contains = _Variantish_state::_Nothing;
     }
 
+    template <class... _Types>
+    constexpr void _Emplace_first(_Types&&... _Args) noexcept(is_nothrow_constructible_v<_Ty1, _Types...>) {
+        _Clear();
+
+        _Construct_in_place(_First, _STD forward<_Types>(_Args)...);
+        _Contains = _Variantish_state::_Holds_first;
+    }
+
+    template <class... _Types>
+    constexpr void _Emplace_second(_Types&&... _Args) noexcept(is_nothrow_constructible_v<_Ty2, _Types...>) {
+        _Clear();
+
+        _Construct_in_place(_Second, _STD forward<_Types>(_Args)...);
+        _Contains = _Variantish_state::_Holds_second;
+    }
+
     union {
-        _It _Iterator;
-        _Se _Sentinel;
+        _Ty1 _First;
+        _Ty2 _Second;

```

Comment:
I believe that inaccurate names would also cause trouble with debugging `join_with_view`, especially since the iterator returned by `join_with_view::end` holds the first alternative ("_Iterator"), while the iterator returned by `join_with_view::begin` usually holds the second alternative ("_Sentinel").

If renaming these members are infeasible, I would rather change back to use `std::variant`.